### PR TITLE
ci: matrix build for testing

### PR DESCRIPTION
**Pull request by knqyf263**
## Description

- Split builds in the GoReleaser configuration
- Build binaries on each environment

## PRs
- https://github.com/aquasecurity/trivy/pull/4585
    - Merge this PR after #4585 gets merged

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by afdesk** [comment]
Should we add `builds` block only for Linux rpm/deb packages?
```yaml
    # Build IDs for the builds you want to create NFPM packages for.
    # Defaults empty, which means no filtering.
    builds:
      - build
```
https://github.com/aquasecurity/trivy/blob/d52b0b7bc0371003ecfd4ef37f9a3643796b5cbd/goreleaser.yml#L50-L79

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -150,10 +143,22 @@ jobs:
...
+        else
+          echo "id=build" >> $GITHUB_OUTPUT
+        fi
+
     - name: Run GoReleaser
       uses: goreleaser/goreleaser-action@v4
       with:
         version: v1.16.2
-        args: release --skip-sign --snapshot --clean --skip-publish --timeout 90m
+        args: build --snapshot --clean --timeout 90m --id ${{ steps.goreleaser_id.outputs.id }}
```
Comment (afdesk) [inline .github/workflows/test.yaml:162]: I'm not sure about change `release` -> `build` subcommand, but removing `--skip-sign` looks dangerous, isn't it?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -150,10 +143,22 @@ jobs:
...
       env:
         GPG_FILE: "nogpg.key"
```
Comment (afdesk) [inline .github/workflows/test.yaml:164]: should we use a real gpg key here?

**Comment by knqyf263** [comment]
>Should we add builds block only for Linux rpm/deb packages?

Yes, it would be the next task. This PR focuses on splitting builds and speeding up tests.

**Comment by knqyf263** [comment]
@afdesk Addressed comments

**Review by afdesk — APPROVED** [review]
LGTM

**Pull request merged**